swagger.yaml

```
@@ -778,16 +811,74 @@ paths:
           description: "Invalid group ID or user ID provided"
           schema:
             $ref: '#/definitions/ErrorResponse'
+        401:
+          $ref: '#/responses/UnauthorizedError'
         500:
           description: "Invalid request body or unexpected Cognito response"
           schema:
             $ref: '#/definitions/ErrorResponse'
+  /groups-report:
+    get:
+      tags:
+        - Groups
+      summary: "List groups and the membership of the groups"
+      description: "To return a list of :-
+          group description
+          group member user email
+          json by default 
+          or if Request Header Accept 'text/csv' then csv format, consisting of preview team description amd user email. 
+        Notable Consideration :- 
+          Included on report
+            groups acting as a role (i.e. publishers, admin)
+          Excluded from Report
+            groups that are empty of users"
+      security:
+        - Authorization: []
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+        - "text/csv"
+      parameters:
+        - in: header
+          name: Accept:application/json
+          type: string
+          required: false
+          description: "Groups report as JSON"
+        - in: header
+          name: Accept:text/csv
+          type: string
+          required:  false
+          description: "Groups report as CSV"
```

I believe it would be better practice to combine these and use an enum:
```suggestion
        - in: header
          name: Accept
          type: string
          enum:
            - "application/json"
            - "text/csv"
          required: false
          description: "Whether a JSON or CSV report is requested"
```

---

Agree